py/desispec/scripts/reformat_proctables.py

```diff
@@ -0,0 +1,177 @@
+            continue
+
+        unexpected = list(curr_colnames[~expected_cols])
+        missing = list(ptab_cols[~found_cols])
+        print(f"Found the following unexpected columns: {unexpected}")
+        print(f"Found the following missing columns: {missing}")
+
+        ## Solving the only cases I'm currently aware of
+        if 'CAMWORD' in unexpected and 'PROCCAMWORD' in missing:
+            print(f"CAMWORD listed instead of PROCCAMWORD. Updating that.")
+            origtable.rename_column('CAMWORD', 'PROCCAMWORD')
+            unexpected.remove('CAWORD')
+            missing.remove('PROCCAMWORD')
+
+        if len(unexpected) > 0:
+            print(f"WARNING: Script detected unexpected columns. Only handle "
+                  + f"the case where 'CAMWORD' is defined instead of PROCCAMWORD. "
+                  + f"The following unexpected columns will be dropped without "
+                  + f"using the information they contain: {unexpected}.")
+            for colname in unexpected:
+                origtable.remove_column(colname)
+
+        ## Add any missing columns
+        for colname in missing:
+            if colname not in ['BADAMPS', 'LASTSTEP', 'EXPFLAG']:
+                print(f"WARNING: Script didn't expect {colname} to be missing. "
+                      + f"Replacing with default values, but this may have "
+                      + f"downstream consequences.")
+            colindex = np.where(ptab_cols==colname)[0][0]
+            newdat = [ptab_defs[colindex]] * len(origtable)
+            newcol = Table.Column(name=colname, data=newdat, dtype=ptab_dtypes[colindex])
+            origtable.add_column(newcol)
+
+        ## Finally, reorder to the current column ordering
+        origtable = origtable[list(ptab_cols)]
+
+        ## If just testing, print the table and a cell-by-cell equality test
+        ## for the scalar columns
+        ## If not testing, move the original table to an archived filename
+        ## and save the updated table to the official exptable pathname
+        if dry_run:
+            print("\n\nOutput file would have been:")
+            origtable.pprint_all()
+        else:
+            ftime = time.strftime("%Y%m%d_%Hh%Mm")
+            replaced_pathname = orig_pathname.replace(f".{orig_filetype}",
+                                                      f".replaced-{ftime}.{orig_filetype}")
+            print(f"Moving original file from {orig_pathname} to {replaced_pathname}")
+            os.rename(orig_pathname,replaced_pathname)
+            time.sleep(0.1)
+            out_pathname = orig_pathname.replace(f".{orig_filetype}", f".{out_filetype}")
+            write_table(origtable, out_pathname)
+            print(f"Updated file saved to {out_pathname}. Original archived as {replaced_pathname}")
+
+            print("\n\n")
+
+        ## Flush the outputs
+        sys.stdout.flush()
+        sys.stderr.flush()
+    print("Exposure table regenerations complete")
```

Comment:
Out of date print statement (exposure vs. processing table regeneration)

Reply:
Fixed